### DISCUSSION TITLE
Reading ESC Telemetry Data via DSHOT Protocol on Crazyflie Bolt

### DISCUSSION THREAD

**Discussion by antalpeter1** [bitcraze/discussions#1576] [Q&A]
Hello Bitcraze community,

I'm working on integrating telemetry data from an ESC into my project using the Crazyflie Bolt 1.1 and a T-Motor Velox V50A ESC, which supports the DSHOT protocol. My goal is to read motor telemetry such as RPM, current, voltage, and temperature directly through DSHOT communication.

My questions are the following:
- **Telemetry Wiring**: The ESC has a separate telemetry wire. Is there an available pin on the Crazyflie Bolt that can be used for telemetry input? How should it be configured in firmware?
- **Software Configuration**: I'm currently using ONESHOT125 in the firmware. Can I just switch the build config to DSHOT and expect everything to work? 
- What would be the correct steps to enable telemetry and display the data on the Crazyflie client or log it for analysis?

Any guidance on connecting the ESC telemetry output to the Crazyflie Bolt and configuring the system would be greatly appreciated. If anyone has successfully implemented this, I’d love to hear about your experience and any additional tips or potential hurdles to watch out for.

Thanks in advance for your help!
Peter

**Comment by tobbeanton** [comment]
This is something that would be a great addition.

> Telemetry Wiring: The ESC has a separate telemetry wire. Is there an available pin on the Crazyflie Bolt that can be used for telemetry input? How should it be configured in firmware?

I think it would be hard. Do you know what type of telemtery protocol, UART? There are only two availible UARTs on the deck port so that would be hard to implement.

> Software Configuration: I'm currently using ONESHOT125 in the firmware. Can I just switch the build config to DSHOT and expect everything to work?

Yes, but there is no DHOT telemetry functionality implemented.

> What would be the correct steps to enable telemetry and display the data on the Crazyflie client or log it for analysis?

Currently we have [not implemented telemetry](https://github.com/bitcraze/crazyflie-firmware/issues/1411). It would be great to have but could be a quite complex task to implement.
